nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -805,21 +805,22 @@ def screen_matches(last: bool) -> bool:
         with self.nested("waiting for {} to appear on screen".format(regex)):
             retry(screen_matches)
 
-    def wait_for_console_text(self, regex: str) -> None:
+    def wait_for_console_text(self, regex: str) -> re.Match[str]:
         with self.nested("waiting for {} to appear on console".format(regex)):
             # Buffer the console output, this is needed
             # to match multiline regexes.
             console = io.StringIO()
             while True:
                 try:
-                    console.write(self.last_lines.get())
+                    console.write(self.last_lines.get(timeout=1))
```

Comment:
I think this is possibly a breaking change: any reason why you're making it non-blocking?

Reply:
Because otherwise it will wait forever and never run the scheduled callbacks. When it times out it throws `Empty`, which was already caught, so it basically just runs callbacks and then retries (we could probably also get rid of the `sleep` here)

Reply:
(I'm not sure how you imagine this to fail, but I personally don't see it)

Reply:
(just realized `Machine.sleep` also indirectly runs callbacks, so I guess just setting the timeout would be enough here)